### ISSUE TITLE
Hand keypoint extraction

### ISSUE THREAD

**Issue by AndrGolubkov** [google/mediapipe#64]
Is it possible using the Hand Tracking (GPU) example to extract not an video, but an array of keypoints? Perhaps I didn’t carefully read the documentation and considered the example, I apologize in advance.

**Comment by ajinkyapuar** [comment]
> Is it possible using the Hand Tracking (GPU) example to extract not an video, but an array of keypoints? Perhaps I didn’t carefully read the documentation and considered the example, I apologize in advance.

https://www.tensorflow.org/lite/guide/inference

**Comment by AndrGolubkov** [comment]
@ajinkyapuar Yes, I noticed that the models are available. But here the question is precisely in obtaining an array of coordinates instead of the rendered output

**Comment by MedlarTea** [comment]
@AndrGolubkov  It can't output precise coordinates!! You can see the graph below carefully:
https://mediapipe.readthedocs.io/en/latest/hand_tracking_mobile_gpu.html
You can find that the 2D keypoints location output are based uv-coordinate.
And in the file /mediapipe/tflite/tflite_tensors_to_landmarks_calculator.proto, you will find that the output z-coordinate is normalized, which don't have real scale.

**Comment by fanzhanggoogle** [comment]
> @ajinkyapuar Yes, I noticed that the models are available. But here the question is precisely in obtaining an array of coordinates instead of the rendered output

If you look at the [hand tracking graph](https://mediapipe.readthedocs.io/en/latest/hand_tracking_mobile_gpu.html) as @MedlarTea  mentioned, you can find the landmarks are output from the HandLandmarkSubgraph at stream "LANDMARKS:hand_landmarks".
You can find the definition of landmark [here](https://github.com/google/mediapipe/blob/master/mediapipe/framework/formats/landmark.proto). 
And @MedlarTea is correct about the scale that the output landmarks is in the image coordinates.

**Comment by chuoling** [comment]
@AndrGolubkov 
Are you asking about getting the landmark coordinates in C++ (e.g., to be used in another calculator), or getting them in Android to be consumed in the Android application?

**Comment by astinmiura** [comment]
@chuoling 
How to get "LANDMARKS:hand_landmarks" in Android?

I ran below code, becuase "LANDMARKS:hand_landmarks" is vector of proto, but failed.
```java
processor.getGraph().addPacketCallback("hand_landmarks", new PacketCallback() {
  @Override
  public void process(Packet packet) {
    PacketGetter.getVectorOfPackets(packet);
  }
});
```
And I think a function to get type of packet is necessary.

**Comment by AndrGolubkov** [comment]
@chuoling I am interested in using and getting key points on iOS.

**Comment by oishi89** [comment]
I have the same question. But I'm wondering if the input is a 2d image then it's so hard to extract a 3D coordinator. Unless the input is a depth image containing depth data.

**Comment by metalwhale** [comment]
@MedlarTea 
But theoretically, it is possible to precisely extract keypoints by using [hand landmark model file](https://github.com/google/mediapipe/blob/master/mediapipe/models/hand_landmark.tflite) combined with `MediaPipe`, isn't it?
I mean, if this was not possible, so why the rendered video can denote those landmarks so exactly like [that](https://github.com/google/mediapipe/raw/master/mediapipe/docs/images/mobile/hand_tracking_android_gpu.gif)?

**Comment by fanzhanggoogle** [comment]
> I have the same question. But I'm wondering if the input is a 2d image then it's so hard to extract a 3D coordinator. Unless the input is a depth image containing depth data.

Hi @oishi89 ,
The model takes in RGB only and output 3D coordinates. We trained out model jointly with synthetic data which has 3D coordinates and the model was able to generalize the z-coord to real images (although it's not perfect yet, we are actively working on it). You can read the Hand Landmark Model session in our [blogpost](https://ai.googleblog.com/2019/08/on-device-real-time-hand-tracking-with.html) for more detail.

**Comment by chuoling** [comment]
@AndrGolubkov @astinmiura 
We'll look into how to best access such information in the iOS/Java API, and provide an example in the next release.

**Comment by AndrGolubkov** [comment]
@chuoling Thank you very much, that would be great

**Comment by Hemanth715** [comment]
@chuoling Thank you, we were hoping for such an API when we first read about this project. We all would appreciate this.

**Comment by mgyong** [comment]
@Hemanth715 @AndrGolubkov @astinmiura Before such an API is available, we have an intermediate solution in C++. See issue from #200 where we have [example of Normalizedlandmark protos](https://github.com/google/mediapipe/issues/200#issuecomment-549168443)

**Comment by mgyong** [comment]
@AndrGolubkov @astinmiura Fixed in [v0.6.6](https://github.com/google/mediapipe/releases/tag/v0.6.6) Pls check it out and let us know

**Comment by faizahmed618** [comment]
Is there any way to extract the keypoint in python so that i can use these in the VR project ? Thankyou :)
